File: src/src_common/fmg/common/geom/PointDouble.cs
using System;

namespace fmg.common.geom {

   public struct PointDouble {
      public double X, Y;

      //public PointDouble() { X=Y=0; }
      public PointDouble(double x, double y) { X = x; Y = y; }
      public PointDouble(PointDouble p) { X = p.X; Y = p.Y; }
      public PointDouble(Point p) { X = p.X; Y = p.Y; }

      public static bool operator !=(PointDouble p1, PointDouble p2) { return !p1.X.HasMinDiff(p2.X) || !p1.Y.HasMinDiff(p2.Y); }
      public static bool operator ==(PointDouble p1, PointDouble p2) { return  p1.X.HasMinDiff(p2.X) &&  p1.Y.HasMinDiff(p2.Y); }

      public override bool Equals(object other) {
         if (ReferenceEquals(null, other))
            return false;
         return (other is PointDouble) && (this == (PointDouble)other);
      }
      public override int GetHashCode() {
         var bits = BitConverter.DoubleToInt64Bits(X);
         bits ^= BitConverter.DoubleToInt64Bits(Y) * 31;
         return (((int)bits) ^ ((int)(bits >> 32)));
      }

      public override string ToString() {
         return "{x:" + X + ", y:" + Y + "}";
      }

      public PointDouble Move(SizeDouble s) { return Move(s.Width, s.Height); }
      public PointDouble Move(double w, double h) {
         X += w;
         Y += h;
         return this;
      }
   }


   public static class PointDoubleExt {
#if WINDOWS_RT || WINDOWS_UWP
      public static PointDouble ToFmRectDouble(this Windows.Foundation.Point self) { return new PointDouble(self.X, self.Y); }
      public static Windows.Foundation.Point ToWinPoint(this PointDouble self) { return new Windows.Foundation.Point { X = self.X, Y = self.Y }; }
#elif WINDOWS_FORMS
      ...
#endif
   }

}

File: src/src_common/fmg/common/geom/PointDouble.java
package fmg.common.geom;

public class PointDouble {
   public double x, y;

   public PointDouble() { x = y = 0;    }
   public PointDouble(double x, double y) { this.x = x; this.y = y; }
   public PointDouble(PointDouble p) { this.x = p.x; this.y = p.y; }

   @Override
   public int hashCode() {
      long temp = Double.doubleToLongBits(x);
      int result = 31 + (int) (temp ^ (temp >>> 32));
      temp = Double.doubleToLongBits(y);
      return 31 * result + (int) (temp ^ (temp >>> 32));
   }
   
   @Override
   public boolean equals(Object obj) {
      if (this == obj) return true;
      if (!(obj instanceof PointDouble))
         return false;
      PointDouble other = (PointDouble) obj;
      return (Double.doubleToLongBits(x) == Double.doubleToLongBits(other.x)) &&
             (Double.doubleToLongBits(y) == Double.doubleToLongBits(other.y));
   }

   @Override
   public String toString() { return "{ x:" + x + ", y:" + y + " }"; }

}
